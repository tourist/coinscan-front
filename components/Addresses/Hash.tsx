import Typography from '@mui/material/Typography';

type HashProps = {
  text: string;
  short?: boolean;
  title?: string;
};
const Hash = ({ text, title, short }: HashProps) => (
  <Typography
    title={title || text}
    sx={{
      textOverflow: short ? 'ellipsis' : null,
      overflow: 'hidden',
      maxWidth: short ? '220px' : null,
      marginRight: 1,
    }}
    variant="body2"
  >
    {text}
  </Typography>
);

export default Hash;
