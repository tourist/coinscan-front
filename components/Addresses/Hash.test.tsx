import { screen, render } from '@testing-library/react';
import Hash from './Hash';

test('render hash', async () => {
  const { container } = render(
    <Hash text="0x320a50f32fb9e20fe113573031132c89835e496c" />
  );
  screen.getByText('0x320a50f32fb9e20fe113573031132c89835e496c');
  expect(container).toMatchSnapshot('render while mouse over icon');
});
