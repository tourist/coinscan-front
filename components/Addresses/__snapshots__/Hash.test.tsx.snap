// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render hash: render while mouse over icon 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body2 css-rdc7yt-MuiTypography-root"
    title="0x320a50f32fb9e20fe113573031132c89835e496c"
  >
    0x320a50f32fb9e20fe113573031132c89835e496c
  </p>
</div>
`;
